packages/http/src/mocker/negotiator/NegotiatorHelpers.ts

```diff
@@ -23,6 +24,42 @@ import {
 } from './InternalHelpers';
 import { IHttpNegotiationResult, NegotiatePartialOptions, NegotiationOptions } from './types';
 
+const warnMsg = (contentTypes: string[]) => `Unable to find content for ${contentTypes}`;
+
+const optionallyGetPayloadlessResponse = (
+  response: IHttpOperationResponse,
+  headers: IHttpHeaderParam[],
+  mediaTypes: string[],
+): Option.Option<IHttpNegotiationResult> => {
+  return pipe(
+    Option.fromNullable(response.contents),
+    Option.map(contents =>
+      contents.filter(c => {
+        return !(c.mediaType === '*/*');
+      }),
+    ),
+    Option.map(contents =>
+      contents.filter(c => {
+        return !c.schema && !(c.examples || []).length;
+      }),
+    ),
+    Option.chain(responses => {
+      return Option.fromPredicate((noContentResponses: { length: number }) => {
+        const noAcceptAndNoContentResponses =
+          !noContentResponses.length && !mediaTypes.filter((mt: string) => !mt.includes('*/*')).length;
+
+        return noAcceptAndNoContentResponses || ['400', '422'].includes(response.code);
```

Comment:
Why are you checking for 400 and 422 responses manually? Don't they follow the same flow?